tool/example.dx.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:butterfly/butterfly.dart';
+
+class FooWidget extends StatelessWidget {
+  @override
+  Node build() {
+    var local = false;
+    var bar = 2;
+    var items = [1, 2, 3].map((x) => Dx('''"${x}"''')).toList();
+
+    return Dx('''
+      <MyComponent foo=${bar}>
+        "This is text"
```

Comment:
Lets come back to text nodes, there are definitely a few different ways we can solve it.  For self closing nodes I am treating them just as 'I have no children'